packages/@ourworldindata/grapher/src/tooltip/TooltipContents.tsx

```diff
@@ -0,0 +1,262 @@
+                </span>
+            </Variable>
+        )
+    }
+}
+
+const includesInsensitive = (str?: string, fragment?: string): boolean =>
+    !!str &&
+    !!fragment &&
+    str.toLocaleLowerCase().includes(fragment.toLocaleLowerCase())
+
+class Variable extends React.Component<{
+    column: CoreColumn
+    color?: string
+    notice?: number | string
+    children?: React.ReactNode
+}> {
+    render(): JSX.Element | null {
+        const { column, children, color, notice } = this.props
+        if (column.isMissing || column.name == "time") return null
+
+        // use the `unit` if it's defined, otherwise look for a parenthetical
+        // at the end of the `displayName` and treat that as the unit
+        const { unit, shortUnit, displayName: fullName } = column,
+            [_m, baseName, parentheticalUnit] =
+                fullName.match(/^(.*?)(?:\s*\(([^\(\=]*?)\))?$/) ?? [],
+            longUnit =
+                unit && unit != shortUnit
+                    ? unit.replace(/^\((.*)\)$/, "$1")
+                    : undefined,
+            displayUnit =
+                longUnit && !includesInsensitive(fullName, longUnit)
+                    ? longUnit
+                    : parentheticalUnit,
+            displayName =
+                parentheticalUnit && displayUnit === parentheticalUnit
+                    ? baseName
+                    : fullName,
+            noticeSpan = notice && (
+                <span className="time-notice">
+                    <FontAwesomeIcon icon={faInfoCircle} />
+                    {notice}
+                </span>
+            )
+
+        return (
+            <div className="variable">
+                <div className="definition">
+                    {displayName && <span className="name">{displayName}</span>}
+                    {displayUnit && displayUnit.length > 1 && (
+                        <span className="unit">{displayUnit}</span>
+                    )}
+                </div>
+                <div className="values" style={{ color }}>
+                    {children}
+                    {noticeSpan}
+                </div>
+            </div>
+        )
+    }
+}
+
+export class TooltipTable extends React.Component<TooltipTableProps> {
+    render(): JSX.Element | null {
+        const { columns, totals, rows } = this.props,
+            focal = rows.some((row) => row.focused),
+            swatched = rows.some((row) => row.swatch),
+            format = { trailingZeroes: true, ...this.props.format },
+            tooEmpty =
+                rows.length < 2 ||
+                totals?.every((value) => value === undefined) ||
+                rows.some(({ values }) =>
+                    values.every((value) => value === undefined)
+                ),
+            tooTrivial = zip(columns, totals ?? []).every(
+                ([column, total]) =>
+                    !!column?.formatValueShort(total).match(/^100(\.0+)?%/)
+            )
```

Comment:
This comes down to preference, but I find many comma-separated statements difficult to read. Probably because my brain tricks me into thinking they're related.